### DISCUSSION TITLE
Custom deepedit and previously trained deepgrow models

### DISCUSSION THREAD

**Discussion by sabino-ramirez** [Project-MONAI/MONAILabel#868] [Q&A] [answered]
Hello all, 

MONAI Label has been great and I can definitely see the potential. However, there are some things that are a little unclear for me so far.

When training a deepedit model on our own data, is it necessary to add a background label? I'm unsure if the background label has to be explicitly annotated, or if for instance the model will accept labels, say 1-3, and everything else is considered '0' for background.
Also, just for peace of mind, do I have to add background labels when training a deepgrow model using Monai Label? 

Lastly, how can I integrate a Deepgrow 3d model trained using clara train (MMAR) into Monai Label? I have tried using the steps available in the sample apps folder of the github https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology#how-to-add-new-model to no avail. 

Thanks for reading!

**Comment by diazandr3s** [comment, marked as answer]
Hi @sabino-ramirez,


Thanks for opening this discussion. 

> When training a deepedit model on our own data, is it necessary to add a background label? I'm unsure if the background label has to be explicitly annotated, or if for instance the model will accept labels, say 1-3, and everything else is considered '0' for background.
> Also, just for peace of mind, do I have to add background labels when training a deepgrow model using Monai Label?

This is a great question! The main reason for having the background label is because DeepEdit simulates clicks for what is background. In this case, the background label is considered a label as the others. This is more clear for multilabel case.

For the multiple single-label segmentation (DeepGrow), you don't need to add the background label. In this case, the background label is whatever is not the positive label (i.e. liver, spleen, etc) https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/deepgrow_3d.py#L34

I know this may be a bit confusing. We're open to hearing recommendations to improve :)

> Lastly, how can I integrate a Deepgrow 3d model trained using clara train (MMAR) into Monai Label? I have tried using the steps available in the sample apps folder of the github https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology#how-to-add-new-model to no avail.


If I'm not wrong, the current DeepGrow 3D  pretrained model in MONAI Label is the one you find in the MMAR. @finalelement please correct me if I'm wrong :)

Hope this helps

**Reply by sabino-ramirez** [reply]
Hi @diazandr3s thanks for the response.

Sounds good. I just want to be sure before I commit to labeling. I make labels using threshold in segment editor. Does this mean then that I have to "paint" the entire image (outside of other labels of course) and submit that as label 0 for background? Or is it enough to manually annotate only the target structures and simply add a label 0 without actually annotating the background? 

I assumed they were the same model. I would just retrain using ML but I don't have access to the same gpu that was capable. The model I have works well using Nvidia slicer plugin but not with the ML environment. I will keep trying.

**Reply by diazandr3s** [reply]
Hi @sabino-ramirez,

> Does this mean then that I have to "paint" the entire image (outside of other labels of course) and submit that as label 0 for background? Or is it enough to manually annotate only the target structures and simply add a label 0 without actually annotating the background?

ooh no, you don't need to do that. The background label I explained before is just part of the training process. 

During the labeling process, it is enough to manually annotate only the target structures.

**Reply by sabino-ramirez** [reply]
@diazandr3s got it. That's what I thought but I saw a demonstration where you labeled the background using grow from seeds and that threw me off. 
Thanks for responding!
